### PR TITLE
Permit page table static to be updated

### PR DESCRIPTION
**Pull request by jul-sh**

### CODE REVIEW

File: oak_restricted_kernel/src/mm/page_tables.rs [hunk 1]
```diff
@@ -267,3 +267,35 @@ impl Translator for RootPageTable {
...
+    pub const fn empty() -> Self {
+        Self { inner: None }
+    }
+
+    /// Replaces the current pagetable with the parameter, returning the old
+    /// pagetable if there was one. Does not issue the instructions update the
+    /// page table on the CPU level.
+    pub fn replace(&mut self, new_pt: RootPageTable) -> Option<RootPageTable> {
+        self.inner.replace(new_pt)
+    }
```
Comment (jul-sh) [inline oak_restricted_kernel/src/mm/page_tables.rs:286]: This isn't used yet, but will be in the future

File: oak_restricted_kernel/src/mm/page_tables.rs [hunk 2]
```diff
@@ -267,3 +267,35 @@ impl Translator for RootPageTable {
         self.inner.translate_physical_frame(frame)
     }
 }
+
+/// Wrapper struct that holds the current page table is there one.
+pub struct CurrentPageTable {
```
Comment (conradgrobler) [inline oak_restricted_kernel/src/mm/page_tables.rs:272]: nit: perhaps emphasize the fact that this is the current root page table in the name (e.g. CurrentRootPageTable)

File: oak_restricted_kernel/src/mm/page_tables.rs [hunk 3]
```diff
@@ -267,3 +267,35 @@ impl Translator for RootPageTable {
...
+    inner: Option<RootPageTable>,
+}
+
+impl CurrentPageTable {
```
Comment (conradgrobler) [inline oak_restricted_kernel/src/mm/page_tables.rs:276]: This doesn't seem to add much over just using `Option<RootPageTable`, apart from changing the names of some functions. Do you plan on adding more functionality to this wrapper in future?

**Review by conradgrobler — APPROVED** [review]

File: oak_restricted_kernel/src/mm/page_tables.rs (hunk 3 above)
Comment (jul-sh) [reply oak_restricted_kernel/src/mm/page_tables.rs:276]: I would expect it to later on update the pagetable on the CPU level upon updating. 

The other difference is that it cannot be set back to `None`.

**Pull request merged**